src/components/CodeSnippet/CodeSnippetBlock.tsx

```diff
@@ -59,13 +59,22 @@ export default function CodeSnippetBlock({
     appearance === CodeSnippetBlockAppearance.LINUX_PROMPT ||
     appearance === CodeSnippetBlockAppearance.WINDOWS_PROMPT ||
     appearance === CodeSnippetBlockAppearance.URL;
-  let numberedCode: JSX.Element[];
+  let numberedCode: ReactNode;
 
   if (isNumbered) {
     className += "--numbered";
 
     // wrap code lines in spans (and preserve the whitespace)
-    const lines = code.split(/\r?\n/);
+    let lines: ReactNode[];
+    if (Array.isArray(code)) {
```

Comment:
Not sure how necessary this is, but maybe we can add an example in docs for this?

Reply:
Good idea. I don't think it's totally obvious how to use it, especially with numbered lines, so I've added an example.